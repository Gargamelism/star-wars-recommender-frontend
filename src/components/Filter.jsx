import React from 'react';

export default function Filter ({ setFilterVal, placeholder = '' }) {
    
    const handleFilterInput = (event) => {
        event.preventDefault();

        setFilterVal(event.target.value);
    };

    return (
        <input 
            type="text" 
            id="characters-filter" 
            className="w-9/12 rounded-lg border-black border-2" 
            onChange={handleFilterInput} 
            placeholder={placeholder}
        />
    );
}