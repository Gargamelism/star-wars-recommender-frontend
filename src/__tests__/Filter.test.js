import renderer from "react-test-renderer";
import Filter from "../components/Filter";

test("renders Filter", () => {
  const tree = renderer
    .create(<Filter setFilterVal={(val) => (val)} placeholder="this place" />)
    .toJSON();
    
    expect(tree).toMatchSnapshot();  
});
