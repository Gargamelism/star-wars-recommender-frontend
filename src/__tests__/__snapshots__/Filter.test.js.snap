// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Filter 1`] = `
<input
  className="w-full rounded-lg border-black border-2 mr-0 md:mr-2 mb-2 md:mb-0"
  id="characters-filter"
  onChange={[Function]}
  placeholder="this place"
  type="text"
/>
`;
